src/NotFoundException.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Http\Discovery;
+
+@trigger_error('The '.__NAMESPACE__.'\NotFoundException class is deprecated since version 1.0 and will be removed in 2.0. Use Http\Discovery\Exception\NotFoundException instead.', E_USER_DEPRECATED);
```

Comment:
i think as long as we ourselves throw the old exception, we can't trigger a warning.

RE: namespace or not: does not matter imo. writing the actual namespace seems more straightforward though